skbio/alignment/_tabular_msa.py

```diff
@@ -570,3 +549,61 @@ def reindex(self, key=None, keys=None):
             keys_.flags.writeable = False
 
         self._keys = keys_
+
+    @experimental(as_of='0.4.0-dev')
+    def append(self, sequence):
+        """Append a sequence to the MSA.
+
+        Parameters
+        ----------
+        sequence : IUPACSequence
+            Sequence to be appended. Must match the dtype of the MSA and the
+            length of the second dimension of the MSA.
+
+        Raises
+        ------
+        TypeError
+            If the type of the sequence does not match the dtype of the MSA.
+        ValueError
+            If the length of the sequence does not match the length of the
+            second dimension of the MSA.
+
+        Notes
+        -----
+        The MSA is not automatically re-aligned when a sequence is appended.
+        Therefore, this operation is not necessarily meaningful on its own.
+
+        Examples
+        --------
+        >>> from skbio import DNA, TabularMSA
+        >>> msa = TabularMSA([DNA('ACG'), DNA('AC-')])
+        >>> len(msa)
+        2
+        >>> msa.append(DNA('AAA'))
+        >>> len(msa)
+        3
+
+        """
+        msa_is_empty = len(self) == 0
+        if msa_is_empty:
+            dtype = type(sequence)
+            if not issubclass(dtype, IUPACSequence):
+                raise TypeError(
+                    "`sequences` must contain scikit-bio sequence objects "
+                    "that have an alphabet, not type %r" % dtype.__name__)
+            self._dtype = dtype
+            self._shape = _Shape(sequence=1, position=len(sequence))
+            self._seqs = [sequence]
+        elif type(sequence) != self.dtype:
+            raise TypeError(
+                "`sequences` cannot contain mixed types. Type %r does "
+                "not match type %r" %
+                (type(sequence).__name__, self.dtype.__name__))
+        elif len(sequence) != self.shape.position:
+            raise ValueError(
+                "`sequences` must contain sequences of the same "
```

Comment:
Can you please update the error messages in this method so that they make sense for `append`? They're still phrased as they were in the constructor, where we were passed an iterable of `sequences`.